query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -617,4 +625,10 @@ private void validateQueryRequest(ExecutionContext executionContext, QueryReques
           "If distinct selections are requested, there should be no groupBys or aggregations.");
     }
   }
+
+  private String updateValForAvgRate(String val){
+    Double doubleVal = Double.parseDouble(val);
+    Double secondInMilli = 1000.0;
```

Comment:
Rate isn't always 1s is it?? I'm pretty sure we pass the denominator as an argument. Also, please use time objects like `Duration.ofSeconds(1)` to improve readability.

Reply:
We pass the denominator as an argument but when we debugged an issue (https://github.com/hypertrace/hypertrace/issues/285), it came out that the denominator was always 1s in millis.

Reply:
I'm not sure if that's the same issue or not, but we should be respecting the arg passed in, if we're not it's a bug.

Reply:
CC @kotharironak

Reply:
@sarthak77 we do use the passed argument of aggregation function (avg_rate). See here - https://github.com/hypertrace/gateway-service/blob/5c5464dc829b5365a475c0d6ebcbc156f882d466/gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/util/ArithmeticValueUtil.java#L64
The thing, we hardcoded is that, we do it for per second rate.

Reply:
Yes @kotharironak. I meant o say that only. 😅